xarray/core/treenode.py

```diff
@@ -44,6 +45,73 @@ def __init__(self, *pathsegments):
 Tree = TypeVar("Tree", bound="TreeNode")
 
 
+class Children(Mapping[str, Tree], Generic[Tree]):
+    """
+    Dictionary-like container for the immediate children of a single DataTree node.
+
+    This collection can be passed directly to the :py:class:`~xarray.DataTree` constructor via its `children` argument.
+    """
+
+    _treenode: Tree
+
+    # TODO add slots?
+    # __slots__ = ("_data",)
+
+    def __init__(self, treenode: Tree):
+        self._treenode = treenode
+
+    @property
+    def _names(self) -> list[str]:
+        return list(self._treenode._children.keys())
```

Comment:
Why not directly use the set-like DictKeys object?
```suggestion
    def _names(self) -> Set[str]:
        return self._treenode._children.keys()
```

Reply:
I can probably use `.keys()` you're right, but I don't think I should type this as returning `Set`, because I originally tried `return set(self._treenode._children)` and that failed intermittently due to ordering differences.

Reply:
I was thinking of `collections.abc.Set`, which is an abstract base class that includes DictKeys, not the specific built-in `set` class